superset-frontend/src/components/Chart/ChartContextMenu/ChartContextMenu.tsx

```diff
@@ -108,12 +111,12 @@ const ChartContextMenu = (
 
   const showDrillToDetail =
     isFeatureEnabled(FeatureFlag.DRILL_TO_DETAIL) &&
-    canExplore &&
+    (canExplore || canViewDrill) &&
     isDisplayed(ContextMenuItem.DrillToDetail);
 
   const showDrillBy =
     isFeatureEnabled(FeatureFlag.DRILL_BY) &&
-    canExplore &&
+    (canExplore || canViewDrill) &&
```

Comment:
In the drill by modal, there's an "Edit chart" button which redirects to Explore. I believe it should be disabled if `canViewDrill` is true but `canExplore` isn't (with some tooltip like "insufficient permissions")

Reply:
I think in general if can_explore is not found in the permissions, this button should be disabled independent of `can_view_drill_dashboard`

Reply:
Yeah that's what I mean 😄 It wasn't there before because user couldn't open that modal anyways without `can_explore` permission

Reply:
Right makes sense. Thanks for spotting that!